src/main/java/net/dv8tion/jda/api/JDABuilder.java

```diff
@@ -477,14 +515,69 @@ public JDABuilder setRelativeRateLimit(boolean enable)
      * @return The JDABuilder instance. Useful for chaining.
      *
      * @see    #setDisabledCacheFlags(EnumSet)
+     *
+     * @deprecated We add CacheFlags to the enum over time which will be disabled when using this method.
+     *             This introduces breaking changes due to the way the setter works.
+     *             You should use {@link #enableCache(Collection)} and {@link #disableCache(Collection)} instead,
+     *             to disable and enable cache flags without side-effects that may break in future versions.
      */
     @Nonnull
+    @Deprecated
+    @ReplaceWith("enableCache(flags) and disableCache(flags)")
+    @DeprecatedSince("4.2.0")
     public JDABuilder setEnabledCacheFlags(@Nullable EnumSet<CacheFlag> flags)
     {
         this.cacheFlags = flags == null ? EnumSet.noneOf(CacheFlag.class) : EnumSet.copyOf(flags);
         return this;
     }
 
+    /**
+     * Enable specific cache flags.
+     * <br>This will not disable any currently set cache flags.
+     *
+     * @param  flags
+     *         The {@link CacheFlag CacheFlags} to enable
+     *
+     * @throws IllegalArgumentException
+     *         If provided with null
+     *
+     * @return The JDABuilder instance. Useful for chaining.
+     *
+     * @see    #disableCache(Collection)
+     */
+    @Nonnull
+    public JDABuilder enableCache(@Nonnull Collection<CacheFlag> flags)
+    {
+        Checks.noneNull(flags, "CacheFlags");
+        cacheFlags.removeAll(flags);
+        return this;
+    }
+
+    /**
+     * Enable specific cache flags.
+     * <br>This will not disable any currently set cache flags.
+     *
+     * @param  flag
+     *         {@link CacheFlag} to enable
+     * @param  flags
+     *         Other flags to enable
+     *
+     * @throws IllegalArgumentException
+     *         If provided with null
+     *
+     * @return The JDABuilder instance. Useful for chaining.
+     *
+     * @see    #disableCache(CacheFlag, CacheFlag...)
+     */
+    @Nonnull
+    public JDABuilder enableCache(@Nonnull CacheFlag flag, @Nonnull CacheFlag... flags)
```

Comment:
Would it be worth it to point this at the `Collections` variant method?

Reply:
You mean an `@see`?